extmod/modussl_mbedtls.c

```diff
@@ -191,7 +193,11 @@ STATIC mp_obj_ssl_socket_t *socket_new(mp_obj_t sock, struct ssl_args *args) {
         goto cleanup;
     }
 
-    mbedtls_ssl_conf_authmode(&o->conf, MBEDTLS_SSL_VERIFY_NONE);
+
+    int certreq = (args->cert_reqs.u_int);
+    mbedtls_ssl_conf_authmode(&o->conf, certreq);
```

Comment:
since `certreq` is not reused, cleaner to write `mbedtls_ssl_conf_authmode(&o->conf, args->cert_reqs.u_int);`

Reply:
Good catch - [updated](https://github.com/daviesian/micropython/blob/ssl-certificate-verification/extmod/modussl_mbedtls.c#L196)!